### PR TITLE
feat(frontend): catalog add properties for ttl

### PR DESCRIPTION
**Pull request by Li0k**
## What's changed and what's your intention?
catalog add properties for

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:
- add properties in catalog.proto for persisting
- handler fit properties
- fix some test with properties

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

## Refer to a related PR or issue link (optional)
- https://github.com/singularity-data/risingwave/issues/3379
- https://github.com/singularity-data/risingwave/issues/3298

### CODE REVIEW

File: src/frontend/src/handler/create_mv.rs [hunk 1]
```diff
@@ -63,19 +67,41 @@ pub fn gen_create_mv_plan(
     let mut table = materialize.table().to_prost(schema_id, database_id);
     let plan: PlanRef = materialize.into();
     table.owner = session.user_name().to_string();
+    table.properties = properties;
 
     Ok((plan, table))
 }
 
+fn handle_create_mv_with_properties(options: Vec<SqlOption>) -> Result<HashMap<String, String>> {
```
Comment (BowenXiao1999) [inline src/frontend/src/handler/create_mv.rs:75]: Actually this function can not only applys to MV?

File: proto/catalog.proto [hunk 2]
```diff
@@ -60,6 +60,7 @@ message Table {
   bool appendonly = 14;
   string owner = 15;
   common.ParallelUnitMapping mapping = 16;
+  map<string, string> properties = 17;
```
Comment (BowenXiao1999) [inline proto/catalog.proto:63]: This is funny. I thought We already have this....

File: src/frontend/src/handler/create_mv.rs [hunk 3]
```diff
@@ -63,19 +67,41 @@ pub fn gen_create_mv_plan(
...
     let session = context.session_ctx.clone();
 
     let (table, graph) = {
-        let (plan, table) = gen_create_mv_plan(&session, context.into(), query, name)?;
+        let (plan, table) = gen_create_mv_plan(
+            &session,
+            context.into(),
+            query,
+            name,
+            handle_create_mv_with_properties(with_options.0)?,
```
Comment (BowenXiao1999) [inline src/frontend/src/handler/create_mv.rs:103]: Why not simply done this inside `gen_create_mv_plan` so we can avoid change the function signature

**Review by BowenXiao1999 — COMMENTED** [review]
Rest LG

**Comment by BowenXiao1999** [comment]
Do we previously use with_option in mv/table?

File: proto/catalog.proto (hunk 2 above)
Comment (Li0k) [reply proto/catalog.proto:63]: yes, it exist in Source , but not Table (

**Review by BowenXiao1999 — APPROVED** [review]

**Review by xiangjinwu — COMMENTED** [review]
Now there are 4 ways to convert a `Vec<SqlOption>` to `HashMap<String, String>` 😂 

* `handle_source_with_properties`
* `handle_create_mv_with_properties`
* `handle_create_table_with_properties`
* `From<WithProperties> for HashMap<String, String>`

**Comment by Li0k** [comment]
> > Now there are 4 ways to convert a Vec to HashMap<String, String> 😂
> > handle_source_with_properties
> > handle_create_mv_with_properties
> > handle_create_table_with_properties
> > From for HashMap<String, String>
> 
> Indeed, I also want to reuse them.

i unify above 4 ways to handle_with_properties.

**Review by xiangjinwu — APPROVED** [review]

**Pull request merged**